modules/GenericDataParser/src/server/generic_data_parser.R

```diff
@@ -706,27 +706,58 @@ validateCalculatedResults <- function(calculatedResults, dryRun, curveNames, tes
         # A project can only be used in the experiment if it's an unrestricted project or belongs to the project which this experiment belongs to
         projectsDT <- data.table(projectDF)
 
-        # Loop through by project code and create a projectAllowedForExperiment column
+        # Add project type if its missing
+        if(!"type" %in% colnames(projectsDT)) {
+          projectsDT[ , type := NA]
+        }
+
+        # Get the project type of the experiment project if required by configurations
+        projectType <- NA
+        if(configList$server.project.type.required) {
+          projectType <- projectsDT[code == projectCode]$type
+          if(is.na(projectType)) {
+            stopUser(paste0("Project '",projectCode,"' does not have a type. Please contact your system administrator."))
+          }
+        }
+
+        # Here we are annotating each of the projects with a boolean as to whether it can be used in the experiment if its used by a lot     
         projectsDT[ , projectAllowedForExperiment := {
-          # Check if the project is unrestricted or if it equals the projectCode of the experiment
-          !isRestricted | code == projectCode
+          # projectType and projectCode are referring to the experiment project
+          # code and isRestricted are referring to the batch project
```

Comment:
I really appreciate these comments. There's one more variable in the mix: `type` which seems to be the batch project type? If that's correct, it would help to add it to this comment

Reply:
Thanks, fixed